paparazzi-gradle-plugin/src/test/projects/compose-recomposition/src/test/java/app/cash/paparazzi/plugin/test/RecomposeTest.kt

```diff
@@ -39,4 +46,36 @@ class RecomposeTest {
       }
     }
   }
+
+  @Test
+  fun customWrappingText(): Unit = paparazzi.snapshot {
+    Column(modifier = Modifier.background(Color.White)) {
+      CustomWrappingText(first = "abc", second = "def")
+      Spacer(modifier = Modifier.size(16.dp))
+      CustomWrappingText(first = "abcd", second = "very long string you should think about it, i like tacos")
+    }
+  }
+}
+
+@Composable
+private fun CustomWrappingText(first: String, second: String) {
+  var textLayoutResult by remember { mutableStateOf<TextLayoutResult?>(null) }
+  var finalText by remember(first, second) { mutableStateOf("$first * $second") }
+  var isSplit by remember(first, second) { mutableStateOf(false) }
+  LaunchedEffect(textLayoutResult) {
+    println("GOT TEXT LAYOUT RESULT ${textLayoutResult != null}")
+    if (textLayoutResult != null && !isSplit && textLayoutResult!!.lineCount > 1) {
+      isSplit = true
+      finalText = "$first\n$second"
+    }
+  }
```

Comment:
@gamepro65 (on vacation, back Friday in ET) I suspect this is related to that LaunchEffect bug you noticed.  Amirite?